doghouse/server/doghouse.go

```diff
@@ -0,0 +1,211 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/bradleyfalzon/ghinstallation"
+	"github.com/google/go-github/github"
+	"github.com/haya14busa/reviewdog"
+	"github.com/haya14busa/reviewdog/diff"
+	"github.com/haya14busa/reviewdog/doghouse"
+)
+
+type DogHouse struct {
+	client *http.Client
+	req    *doghouse.CheckRequest
+	gh     *github.Client
+}
+
+func New(req *doghouse.CheckRequest, privateKey []byte, integrationID int, c *http.Client) (*DogHouse, error) {
+	dh := &DogHouse{client: c, req: req}
+	if err := dh.setGitHubClient(privateKey, integrationID); err != nil {
+		return nil, err
+	}
+	return dh, nil
+}
+
+func (dh *DogHouse) Check(ctx context.Context, req *doghouse.CheckRequest) (*doghouse.CheckResponse, error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method DogHouse.Check should have comment or be unexported